google/cloud/aiplatform/models.py

```diff
@@ -1326,15 +1364,508 @@ def explain(
+            )
+
+        endpoint = cls._empty_constructor(
+            project=project, location=location, credentials=credentials
+        )
+
+        endpoint._gca_resource = gapic_resource
+
+        endpoint._http_client = urllib3.PoolManager()
+
+        return endpoint
+
+    def _http_request(
+        self,
+        method: str,
+        url: str,
+        body: Optional[Dict[Any, Any]] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> "urllib3.response.HTTPResponse":  # type: ignore # noqa: F821
+        """Helper function used to perform HTTP requests for private Endpoint.
+
+        Args:
+            method (str):
+                Required. The HTTP request method to use. Example: "POST" or "GET"
+            url (str):
+                Required. The url used to send requests and get responses from.
+            body (Dict[Any, Any]):
+                Optional. Data sent to the url in the HTTP request. For private
+                Endpoint, an instance is sent and a prediction response is expected.
+            headers (Dict[str, str]):
+                Optional. Header in the HTTP request.
+
+        Returns:
+            urllib3.response.HTTPResponse:
+                A HTTP Response container.
+
+        Raises:
+            RuntimeError: If a HTTP request could not be made.
+        """
+        try:
+            import urllib3
+        except ImportError:
+            raise ImportError(
+                "Cannot import the urllib3 HTTP client. Please install google-cloud-aiplatform[private_endpoints]."
+            )
+
+        try:
+            response = self._http_client.request(
+                method=method, url=url, body=body, headers=headers
+            )
+
+            if response.status < 300:
+                return response
+            else:
+                raise RuntimeError(
+                    f"{response.status} - Failed to make request, see response: "
+                    + response.data.decode("utf-8")
+                )
+
+        except urllib3.exceptions.MaxRetryError as exc:
+            raise RuntimeError(
+                f"Failed to make a {method} request to this URI, make sure: "
+                " this call is being made inside the network this private Endpoint is peered to "
+                f"({self._gca_resource.network}), calling health_check() returns True, "
+                f"and that {url} is a valid URL."
+            ) from exc
+
+    def predict(self, instances: List, parameters: Optional[Dict] = None) -> Prediction:
+        """Make a prediction against this private Endpoint using a HTTP request.
+        This method must be called within the network the private Endpoint is peered to.
+        The predict() call will fail otherwise. To check, use `PrivateEndpoint.network`.
+
+        Example usage:
+            response = my_private_endpoint.predict(instances=[...])
+            my_predictions = response.predictions
+
+        Args:
+            instances (List):
```

Comment:
Maybe make a note that the instance types must be JSON serializable.

Seems there is no clear guideline on how to typehint Serializable. Closest I see is https://www.oreilly.com/library/view/mastering-object-oriented-python/9781789531367/13d1b3d7-12ef-4d21-bd0b-1d45c3ac1ad8.xhtml.

I think including the requirement in the docstring is fine for now.